File: RemsNG.Common/Models/TaxPayerModel.cs
using System;

namespace RemsNG.Common.Models
{
    public class TaxPayerModel
    {
        public Guid Id { get; set; }
        public Guid CompanyId { get; set; }
        public Guid? StreetId { get; set; }
        public Guid? AddressId { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? DateCreated { get; set; }
        public string Lastmodifiedby { get; set; }
        public DateTime? LastModifiedDate { get; set; }
        public string TaxpayerStatus { get; set; }
        public string Surname { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public string companyName { get; set; }
        public string WardName { get; set; }

        public virtual CompanyModel Company { get; set; }
        public virtual StreetModel Street { get; set; }

        public string StreetNumber { get; set; }
        public string StreetName { get; set; }

        public Guid WardId { get; set; }
        public int ItemCount { get; set; }
    }

    public enum TaxPayerEnum
    {
        ACTIVE, DELETED
    }
}

File: RemsNG/RemsNG-UI/src/app/taxpayers/models/taxpayer.model.ts
export class TaxpayerModel{
    id: string = '';
    companyId:string = '';
    streetId: string = '';
    addressid: string = '';
    taxpayerStatus: string = '';
    companyName: string = '';
    dateCreated: string = '';
    createdBy: string = '';
    lastmodifiedby: string = '';
    lastModifiedDate: string = '';
    streetNumber: string = '';
    eventType:string = '';
    isLoading:boolean = false;
    isConfirmCompany:boolean = true;
    surname: string ='';
    firstname: string ='';
    lastname: string ='';
    wardId: string = '';
    isOneTime: boolean = false;    
    isChecked = false;
  
    constructor(data: any){
        this.id = data.id;
        this.companyId = data.companyId;
        this.streetId=data.streetId;
        this.addressid=data.addressid;        
        this.taxpayerStatus = data.taxpayerStatus;
        this.companyName = data.companyName;
        this.dateCreated = data.dateCreated;
        this.createdBy = data.createdBy;
        this.lastmodifiedby= data.lastmodifiedby;
        this.streetNumber= data.streetNumber;
        this.eventType = data.eventType;
        this.surname=data.surname;
        this.firstname=data.firstname;
        this.lastname = data.lastname;
        this.wardId= data.wardId;
        this.isOneTime=data.isOneTime;
        this.lastModifiedDate = data.lastModifiedDate;
    }
}